crates/gosub-bindings/include/nodes.h

```diff
@@ -13,15 +13,23 @@ void render_tree_node_free(struct node_t **node);
 
 enum node_type_e { NODE_TYPE_ROOT = 0u, NODE_TYPE_TEXT };
 
+struct position_t {
+  double x;
+  double y;
+};
+
 struct node_t {
   enum node_type_e type;
+  struct position_t position;
   union data {
     bool root;               // NODE_TYPE_ROOT
     struct node_text_t text; // NODE_TYPE_TEXT
   } data;
 };
 
 struct node_t *render_tree_node_init();
+double render_tree_node_get_x(const struct node_t *node);
+double render_tree_node_get_y(const struct node_t *node);
```

Comment:
Maybe it makes sense to attach these functions to the node_t structs. The only problem I'd see with attaching them to the node_t is, that it could make it harder to call them.

Reply:
What do you mean exactly? These already take a node_t

Reply:
Ah, what i meant is a c++ only feature thought, i did this once in C. Also, i thought i had  seen this once in a c library.

Reply:
Oh, yea c++ allows methods in structs since they’re a special case of a class in that case

Reply:
Some people put function pointers in structs but I only do that if it’s really necessary

Reply:
Ya, I thought this was possible because C had no classes. But probably when c had this, there would be no c++